mat/dense.go

```diff
@@ -201,10 +201,16 @@ func (m *Dense) T() Matrix {
 	return Transpose{m}
 }
 
-// ColView returns a VecDense reflecting the column j, backed by the matrix data.
+// ColView returns a Vector reflecting the column j, backed by the matrix data.
```

Comment:
I think the current documentation is okay, but want to check. With a `*VecDense`, really only Dense could implement the interface. With it being `Vector`, now hypothetically the other types (e.g. `SymDense`) could implement it, and just not have the returned type be mutable. I think "backed by the matrix data" is vague enough to allow this.

Reply:
Agree, `*VecDense` is closely associated to `Dense` so only a `Dense` is likely to return it.  Other `Matrix` types implementing `ColViewer`/`RowViewer` may need to return matrix specific `Vector` implementations, e.g. `SymDense` and `SymDenseVector`, etc.  Let me know if you think the documentation could do with more detail and I can update.